mesonbuild/build.py

```diff
@@ -2595,11 +2595,18 @@ def get_dep_outname(self, infilenames):
                 raise InvalidArguments('Substitution in depfile for custom_target that does not have an input file.')
             return self.depfile
 
+    def is_linkable_output(self, output: str) -> bool:
+        if any(output.endswith(i) for i in {'.a', '.dll', '.lib', '.so', '.dylib'}):
+            return True
+        # libfoo.so.X soname
+        if re.match('.*\.so(\.\d+)*$', output):
```

Comment:
```suggestion
        if re.search(r'\.so(\.\d+)*$', output):
```

Reply:
Assuming it's a type to have dropped `.*`? Done.

Reply:
ah sorry, you changed match to search, makes sense.